docs/eve-images.md

```diff
@@ -9,38 +9,152 @@ to use a different version, you may want to use a different disk image entirely.
+* local qemu, standard distributed eve-os image
+* local qemu, custom eve-os image
+* external device, standard distributed eve-os image
+* external device, custom eve-os image
+
+### QEMU and Standard
+
+This is the most common use case. You run `eden`, which launches EVE in qemu
+running a standard distributed eve-os.
+
+The flow is as follows:
+
+1. `eden config add <name>` - create a named config
+1. `eden config set <name> --key eve.tag --value <tag>` - instruct it to use the desired eve tag from docker.io/lfedge/eve
+1. `eden setup` - in addition to normal eden setup
+    1. pull the image `docker.io/lfedge/eve:<tag>`
+    1. extract the disk image from the docker image
+    1. configure the `config` partition customized for the context
+    1. save the generated disk image to a local cache, normally `$PWD/dist/<name>-images/eve/`
+    1. report to the user the path to the extracted disk image
+1. `eden start` - start EVE in a virtual device via qemu, using the extracted disk as the boot device
+
+### QEMU and Custom
+
+TO have eden tell qemu to run your custom EVE image, you simply need to:
+
+1. Make sure it is saved on the local device as a docker image with a unique tag
+1. Tell eden to use that unique tag
+1. Follow the usual steps of `eden setup` and `eden start`
+
+The flow, then, is as follows:
+
+1. Generate an EVE image in docker image format, saving it to `lfedge/eve:<some custom tag>`
+1. `eden config add <name>` - create a named config
+1. `eden config set <name> --key eve.tag --value <some custom tag>` - instruct it to use the desired eve tag from docker.io/lfedge/eve, which it will find where you placed it, in the local cache
+1. `eden setup` - generate the bootable disk image from the image in the docker cache
+1. `eden start` - start EVE in a virtual device via qemu, using the extracted disk as the boot device
+
+For detailed instructions on how to generate the correct docker image, see
+[Docker Image][Docker Image].
+
+If you do not want to, or cannot, save the EVE image as a docker image, and have
+the bootable image ready, you can do the following:
+
+1. `eden config add <name>` - create a named config
+1. `eden setup` - generate the bootable disk image from the image in the docker cache
+1. `eden start --image-file=path/to/your/live-image` - start EVE in a virtual device via qemu, using the disk file at the provided path as the boot device
+
+Note that in the live image option, `eden setup` _still_ will extract a file and
+configure it, but it will not be used. It is your responsibility to configure
+your custom bootable image as desired, for example controller address and
+certificates.
+
+### External and Standard
+
+This use case is almost identical to [QEMU and Standard][QEMU and Standard]. The
+only differences are:
+
+* we extract the image after `eden setup` to flash it to our device
+* we start our external device on our own
+* we instruct `eden start` to start everything _except_ EVE via qemu
```

Comment:
@giggsoff how? What is it that tells `eden start` _not_ to start qemu?

Reply:
Seems, we have `eve.remote` in our config file, which indicates, that `eden start` should ends before starting of qemu/VB/Parallels. It is set to true with GCP or general models.

Reply:
So `eve.remote=true` means, "the eve is remote and not under eden control, so don't try to start or stop it"?

If so, what is the right invocation to enable it? I _think_ it happens in any of the following cases:

* `eden config add <name> --devmodel general` (automatically sets it)
* `eden config add <name> --devmodel gcp` (automatically sets it)
* `eden config set <name> --key=eve.remote --value=true` (explicitly set it)

Is that correct? Confirm (or correct me), and I will update the PR appropriately.

Reply:
One correction: `GCP`, not `gcp`.  The rest is true

Reply:
When I set `--devmodel general`, is that just shorthand for `eve config set <name> --key <something> --value general`? What is that setting?

Reply:
Or does the actual command `--devmodel` set multiple fields, including both `eve.devmodel` and `eve.remote`?

Reply:
I even added a section. Take another look @giggsoff

Reply:
Yes, `--devmodel general` sets multiple options (https://github.com/lf-edge/eden/blob/master/pkg/models/general.go#L25-L33). Will take a look, thank you!